crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -1016,6 +1032,17 @@ pub enum DiagnosticId {
 
     /// Use of a deprecated setting.
     DeprecatedSetting,
+
+    /// The code needs to be formatted.
+    Unformatted,
+
+    /// Range formatting isn't supported for notebooks.
+    RangeFormatNotebook,
+
+    /// Something went wrong while formatting.
+    ///
+    /// This often indicates a bug in the formatter.
+    FormatError,
```

Comment:
Similar to `RangeFormatNotebook`. I wonder if we should just have one `internal-error` code. The main advantage I see is that it makes it very clear that this is a bug in Ruff vs. some user error (we should probalby add a note saying that too)